core/src/main/java/ai/timefold/solver/core/impl/domain/solution/cloner/FieldAccessingSolutionCloner.java

```diff
@@ -51,14 +57,14 @@ public FieldAccessingSolutionCloner(SolutionDescriptor<Solution_> solutionDescri
 
     @Override
     public @NonNull Solution_ cloneSolution(@NonNull Solution_ originalSolution) {
-        Map<Object, Object> originalToCloneMap = new IdentityHashMap<>();
-        Queue<Unprocessed> unprocessedQueue = new ArrayDeque<>();
-        Solution_ cloneSolution = clone(originalSolution, originalToCloneMap, unprocessedQueue,
+        var originalToCloneMap = CollectionUtils.newIdentityHashMap(EXPECTED_OBJECT_COUNT);
```

Comment:
I wonder, can you make `EXPECTED_OBJECT_COUNT` an instance field, and set it to the actual size of the `originalToCloneMap` after validateCloneSolution?

Reply:
Hmmm, interesting idea. I wonder what the downsides are.

If we first cloned a big solution, and then cloned a small solution, the map would stay big. But If I divided the count by 2, then this problem would be decreased, and paying the price of doubling the array once isn't really much of a cost.

Thoughts?